src/predicate/set.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017, Nick Stevens <nick@bitcurry.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/license/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Definition of `Predicate`s for comparisons of membership in a set.
+
+use std::collections::HashSet;
+use std::hash::Hash;
+use std::iter::FromIterator;
+
+use Predicate;
+
+/// Predicate that returns `true` if `variable` is a member of the pre-defined
+/// `HashSet`, otherwise returns `false`.
+///
+/// This is created by the `predicate::in_set` function.
+#[derive(Debug)]
+pub struct SetPredicate<T>
+    where T: Hash + Eq
+{
+    inner: HashSet<T>,
+}
+
+impl<T> Predicate for SetPredicate<T>
```

Comment:
That is a really good point. There would have to be other restrictions on the implementation`Item` type in order to do intersect, difference, etc like you said, but there's no reason it couldn't be done.

I'll change this type to `ValueInSetPredicate` to make it clear.